File: VB/WeCanFixIt/Controllers/WorkerController.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Linq
Imports System.Threading.Tasks
Imports System.Net
Imports System.Web
Imports System.Web.Mvc
Imports WeCanFixIt.Models

Namespace Controllers
    Public Class WorkerController
        Inherits Controller

        Private ReadOnly _db As New ApplicationDbContext

        ' GET: Worker
        Async Function Index() As Task(Of ActionResult)
            Return View(Await _db.Workers.ToListAsync())
        End Function

        ' GET: Worker/Details/5
        Async Function Details(ByVal id As Integer?) As Task(Of ActionResult)
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim worker As Worker = Await _db.Workers.FindAsync(id)
            If IsNothing(worker) Then
                Return HttpNotFound()
            End If
            Return View(worker)
        End Function

        ' GET: Worker/Create
        Function Create() As ActionResult
            Return View()
        End Function

        ' POST: Worker/Create
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see https://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Async Function Create(<Bind(Include:="Id,Name,Skills,HoursWorked,Rate")> ByVal worker As Worker) As Task(Of ActionResult)
            If ModelState.IsValid Then
                _db.Workers.Add(worker)
                Await _db.SaveChangesAsync()
                Return RedirectToAction("Index")
            End If
            Return View(worker)
        End Function

        ' GET: Worker/Edit/5
        Async Function Edit(ByVal id As Integer?) As Task(Of ActionResult)
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim worker As Worker = Await _db.Workers.FindAsync(id)
            If IsNothing(worker) Then
                Return HttpNotFound()
            End If
            Return View(worker)
        End Function

        ' POST: Worker/Edit/5
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see https://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Async Function Edit(<Bind(Include:="Id,Name,Skills,HoursWorked,Rate")> ByVal worker As Worker) As Task(Of ActionResult)
            If ModelState.IsValid Then
                _db.Entry(worker).State = EntityState.Modified
                Await _db.SaveChangesAsync()
                Return RedirectToAction("Index")
            End If
            Return View(worker)
        End Function

        ' GET: Worker/Delete/5
        Async Function Delete(ByVal id As Integer?) As Task(Of ActionResult)
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim worker As Worker = Await _db.Workers.FindAsync(id)
            If IsNothing(worker) Then
                Return HttpNotFound()
            End If
            Return View(worker)
        End Function

        ' POST: Worker/Delete/5
        <HttpPost()>
        <ActionName("Delete")>
        <ValidateAntiForgeryToken()>
        Async Function DeleteConfirmed(ByVal id As Integer) As Task(Of ActionResult)
            Dim worker As Worker = Await _db.Workers.FindAsync(id)
            _db.Workers.Remove(worker)
            Await _db.SaveChangesAsync()
            Return RedirectToAction("Index")
        End Function

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing) Then
                _db.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
    End Class
End Namespace

File: CS/WeCanFixIt.Web/Controllers/WorkerController.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Entity;
using System.Linq;
using System.Threading.Tasks;
using System.Net;
using System.Web;
using System.Web.Mvc;
using WeCanFixIt.Web.Models;

namespace WeCanFixIt.Web.Controllers
{
    [Authorize]
    public class WorkerController : Controller
    {
        private readonly ApplicationDbContext _db = new ApplicationDbContext();

        // GET: Worker
        public async Task<ActionResult> Index()
        {
            return View(await _db.Workers.ToListAsync());
        }

        // GET: Worker/Details/5
        public async Task<ActionResult> Details(int? id)
        {
            if (id == null)
            {
                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
            }
            Worker worker = await _db.Workers.FindAsync(id);
            if (worker == null)
            {
                return HttpNotFound();
            }
            return View(worker);
        }

        // GET: Worker/Create
        public ActionResult Create()
        {
            ViewBag.SkillList=new SelectList(_db.Jobs, "WorkType", "WorkType");
            return View();
        }

        // POST: Worker/Create
        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        [HttpPost]
        [ValidateAntiForgeryToken]
        public async Task<ActionResult> Create([Bind(Include = "Id,Name,Skills,HoursWorked,Rate")] Worker worker)
        {
            if (ModelState.IsValid)
            {
                _db.Workers.Add(worker);
                await _db.SaveChangesAsync();
                return RedirectToAction("Index");
            }

            ViewBag.SkillList = new SelectList(_db.Jobs, "WorkType", "WorkType");

            return View(worker);
        }

        // GET: Worker/Edit/5
        public async Task<ActionResult> Edit(int? id)
        {
            if (id == null)
            {
                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
            }
            Worker worker = await _db.Workers.FindAsync(id);
            if (worker == null)
            {
                return HttpNotFound();
            }

            ViewBag.SkillList = new SelectList(_db.Jobs, "WorkType", "WorkType");

            return View(worker);
        }

        // POST: Worker/Edit/5
        // To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        // more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        [HttpPost]
        [ValidateAntiForgeryToken]
        public async Task<ActionResult> Edit([Bind(Include = "Id,Name,Skills,HoursWorked,Rate")] Worker worker)
        {
            if (ModelState.IsValid)
            {
                _db.Entry(worker).State = EntityState.Modified;
                await _db.SaveChangesAsync();
                return RedirectToAction("Index");
            }

            ViewBag.SkillList = new SelectList(_db.Jobs, "WorkType", "WorkType");

            return View(worker);
        }

        // GET: Worker/Delete/5
        public async Task<ActionResult> Delete(int? id)
        {
            if (id == null)
            {
                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
            }
            Worker worker = await _db.Workers.FindAsync(id);
            if (worker == null)
            {
                return HttpNotFound();
            }
            return View(worker);
        }

        // POST: Worker/Delete/5
        [HttpPost, ActionName("Delete")]
        [ValidateAntiForgeryToken]
        public async Task<ActionResult> DeleteConfirmed(int id)
        {
            Worker worker = await _db.Workers.FindAsync(id);
            _db.Workers.Remove(worker);
            await _db.SaveChangesAsync();
            return RedirectToAction("Index");
        }

        protected override void Dispose(bool disposing)
        {
            if (disposing)
            {
                _db.Dispose();
            }
            base.Dispose(disposing);
        }
    }
}
